### PR TITLE
Added read_write_int method and unit test for them

### PR DESCRIPTION
**Pull request by injaleea**
[X] This contribution adheres to [CONTRIBUTING.md](https://github.com/ni/<reponame>/blob/master/CONTRIBUTING.md).

### What does this Pull Request accomplish?

Added fake method for read_write_integer attribute
Added unit test for set get integer attribute
unit  test for close with error

### Why should this Pull Request be merged?
More unit test coverage for fake

### What testing has been done?

Passed on dev machine

### CODE REVIEW

File: src/nifake/tests/test_session.py [hunk 1]
```diff
@@ -328,3 +328,39 @@ def test_multiple_return_params(self):
             assert self.patched_errors.handle_error.call_count == 2
             self.patched_errors.handle_error.assert_called_with(session, self.patched_library.niFake_GetCalDateAndTime.return_value)
     '''
+
+    def test_get_vi_int32_attribute(self):
+        self.patched_library.niFake_GetAttributeViInt32.side_effect = self.side_effects_helper.niFake_GetAttributeViInt32
+        int = 3
```
Comment (texasaggie97) [inline src/nifake/tests/test_session.py]: I don't like naming variables the same as the type. Can you call this something different?

File: src/nifake/tests/test_session.py [hunk 2]
```diff
@@ -328,3 +328,39 @@ def test_multiple_return_params(self):
...
+        self.side_effects_helper['GetAttributeViInt32']['attributeValue'] = int
+        with nifake.Session('dev1') as session:
+            attr_int = session.read_write_integer
+            assert(attr_int == int)
+            from mock import call
+            calls = [call(SESSION_NUM_FOR_TEST, b"", 1000004, ANY)]
+            self.patched_library.niFake_GetAttributeViInt32.assert_has_calls(calls)
```
Comment (texasaggie97) [inline src/nifake/tests/test_session.py]: Why does this test use mock.call to check the call, while the next test uses assert_called_once_with()?

File: src/nifake/tests/test_session.py [hunk 3]
```diff
@@ -328,3 +328,39 @@ def test_multiple_return_params(self):
...
+        self.side_effects_helper['close']['return'] = test_error_code
+        self.side_effects_helper['close']['vi'] = SESSION_NUM_FOR_TEST
+        self.patched_library.niFake_GetError.side_effect = self.side_effects_helper.niFake_GetError
+        self.side_effects_helper['GetError']['errorCode'] = test_error_code
+        self.side_effects_helper['GetError']['description'] = test_error_desc
+        session = nifake.Session('dev1')
+        session1 = nifake.Session('dev1')
+        session.close()
+        session1.close()
+        assert True
```
Comment (marcoskirsch) [inline src/nifake/tests/test_session.py]: What is this for?

File: src/nifake/tests/test_session.py [hunk 4]
```diff
@@ -328,3 +328,39 @@ def test_multiple_return_params(self):
...
+    def test_error_on_close(self):
+        test_error_code = -1
+        test_error_desc = 'Test'
+        self.patched_library.niFake_close.side_effect = self.side_effects_helper.niFake_close
+        self.side_effects_helper['close']['return'] = test_error_code
+        self.side_effects_helper['close']['vi'] = SESSION_NUM_FOR_TEST
+        self.patched_library.niFake_GetError.side_effect = self.side_effects_helper.niFake_GetError
+        self.side_effects_helper['GetError']['errorCode'] = test_error_code
+        self.side_effects_helper['GetError']['description'] = test_error_desc
+        session = nifake.Session('dev1')
```
Comment (marcoskirsch) [inline src/nifake/tests/test_session.py]: Why do you need two sessions in this test?

File: src/nifake/tests/test_session.py [hunk 5]
```diff
@@ -328,3 +328,39 @@ def test_multiple_return_params(self):
...
+
+    def test_set_vi_int32_attribute(self):
+        self.patched_library.niFake_SetAttributeViInt32.side_effect = self.side_effects_helper.niFake_SetAttributeViInt32
+        attribute_id = 1000004
+        int = 1
+        with nifake.Session('dev1') as session:
+            session.read_write_integer = int
+            self.patched_library.niFake_SetAttributeViInt32.assert_called_once_with(SESSION_NUM_FOR_TEST, b'', attribute_id, 1)
+
+    def test_error_on_close(self):
```
Comment (marcoskirsch) [inline src/nifake/tests/test_session.py]: One of the things close() should do, is log on failure (cannot / we decided not to raise).

This should have been fixed https://github.com/ni/nimi-python/issues/198 but we missed it.

File: src/nifake/tests/test_session.py [hunk 6]
```diff
@@ -328,3 +328,32 @@ def test_multiple_return_params(self):
...
+            assert(attr_int == test_number)
+            self.patched_library.niFake_GetAttributeViInt32.assert_called_once_with(SESSION_NUM_FOR_TEST, b'', 1000004, ANY)
+
+    def test_set_vi_int32_attribute(self):
+        self.patched_library.niFake_SetAttributeViInt32.side_effect = self.side_effects_helper.niFake_SetAttributeViInt32
+        attribute_id = 1000004
+        test_number = 1
+        with nifake.Session('dev1') as session:
+            session.read_write_integer = test_number
+            self.patched_library.niFake_SetAttributeViInt32.assert_called_once_with(SESSION_NUM_FOR_TEST, b'', attribute_id, 1)
```
Comment (marcoskirsch) [inline src/nifake/tests/test_session.py]: How come you didn't use test_number here?

File: src/nifake/tests/test_session.py [hunk 7]
```diff
@@ -328,3 +328,32 @@ def test_multiple_return_params(self):
...
+
+    def test_error_on_close(self):
+        test_error_code = -1
+        test_error_desc = 'Test'
+        self.patched_library.niFake_close.side_effect = self.side_effects_helper.niFake_close
+        self.side_effects_helper['close']['return'] = test_error_code
+        self.side_effects_helper['close']['vi'] = SESSION_NUM_FOR_TEST
```
Comment (marcoskirsch) [inline src/nifake/tests/test_session.py]: What is this supposed to accomplish? niFake_Close() takes the ViSession as an input. You should be checking that with assert_called_once_with...

File: src/nifake/tests/test_session.py [hunk 8]
```diff
@@ -328,3 +328,32 @@ def test_multiple_return_params(self):
...
+        self.patched_library.niFake_GetError.side_effect = self.side_effects_helper.niFake_GetError
+        self.side_effects_helper['GetError']['errorCode'] = test_error_code
+        self.side_effects_helper['GetError']['description'] = test_error_desc
+        session = nifake.Session('dev1')
+        session.close()
```
Comment (marcoskirsch) [inline src/nifake/tests/test_session.py]: When close() errors, we don't raise an exception. But we do or at least should log a warning (don't remember if that's implemented yet). Can you check that in the test as well?

Comment (marcoskirsch) [reply src/nifake/tests/test_session.py]: #254 is what I was referring to.

File: src/nifake/tests/test_session.py [hunk 9]
```diff
@@ -328,3 +328,32 @@ def test_multiple_return_params(self):
             assert self.patched_errors.handle_error.call_count == 2
             self.patched_errors.handle_error.assert_called_with(session, self.patched_library.niFake_GetCalDateAndTime.return_value)
     '''
+
```
Comment (texasaggie97) [inline src/nifake/tests/test_session.py]: When there are changes to a test file, either system tests or unit tests, there should be a corresponding change in the generated files.

When you don't have the changed generated file as part of your change, it ends up being an unrelated change in a different pull request.
